pkg/build/graph/analysis/bc.go

```diff
@@ -162,17 +162,10 @@ func FindCircularBuilds(g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
 	return markers
 }
 
-// FindPendingTags inspects all imageStreamTags that serve as outputs to builds.
-//
-// Precedence of failures:
-// 1. A build config points to the non existent tag but no current build exists.
-// 2. A build config points to the non existent tag but the latest build has failed.
-func FindPendingTags(g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
-	markers := []osgraph.Marker{}
-
-	for _, uncastIstNode := range g.NodesByKind(imagegraph.ImageStreamTagNodeKind) {
-		istNode := uncastIstNode.(*imagegraph.ImageStreamTagNode)
-		if bcNode := buildedges.BuildConfigForTag(g, uncastIstNode); bcNode != nil && !istNode.Found() {
+// findPendingTags is the guts behind FindPendingTags .... break out some of the contents and reduce some indentation
+func findPendingTags(bcNodes []*buildgraph.BuildConfigNode, markers []osgraph.Marker, istNode *imagegraph.ImageStreamTagNode, uncastIstNode graph.Node, g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
```

Comment:
The number of arguments here smells bad ... almost not sure if we should be passing in `istNode` and `bcNodes` if they're trivial to get from the `uncastIstNode`. 

Similarly, we don't need the `markers`, we can just return our own list and `FindPendingTags` can do the append.